### PR TITLE
fix restart on Linux

### PR DESCRIPTION
**Pull request by tamlok**

### CODE REVIEW

**Comment by ygcaicn** [comment]
Restarting AppImage does not work, which is super confusing. 

image

**Comment by tamlok** [comment]
> Restarting AppImage does not work, which is super confusing.
> 
> image

Maybe we could start a bash like `bash -c "sleep 20 && vnote&" to have a try.

**Comment by ygcaicn** [comment]
I extracted the AppImage and then ran ./usr/bin/vnote, and found the same problem, the parent process could not exit. So it doesn't seem to be a problem with AppImage.

**Comment by tamlok** [comment]
> I extracted the AppImage and then ran ./usr/bin/vnote, and found the same problem, the parent process could not exit. So it doesn't seem to be a problem with AppImage.

Maybe it is caused by the shared memory by the guard? We could try to remove the guard for a test.

**Comment by ygcaicn** [comment]
I have no problems using Qt 5.15.2 locally. In another pr, I tested the use of the atexit function and startedDetached after main exited. This works in the Github workflow. But what happened after main exited causing the process to hang?

**Comment by ygcaicn** [comment]
This is okay.

```cpp   
 if (ret == RESTART_EXIT_CODE) {
        // Asked to restart VNote.
        guard.exit();
        QProcess::startDetached(ConfigMgr::getApplicationFilePath(), QStringList());
        exit(0);
    }
```

File: src/core/configmgr.cpp [hunk 1]
```diff
@@ -378,9 +378,14 @@ QString ConfigMgr::getLogFile() const
...
 {
-#if defined(Q_OS_Linux)
+#if defined(Q_OS_LINUX)
     // TODO: Check if it is from AppImage.
     // We could get the APPIMAGE env variable from the AppRun script and pass it to vnote via cmd.
+    auto appImageVar = QString::fromLocal8Bit(qgetenv("APPIMAGE"));
+    qInfo() << "APPIMAGE" << appImageVar;
+    if (!appImageVar.isEmpty()) {
+        return appImageVar;
+    }
```
Comment (ygcaicn) [inline src/core/configmgr.cpp:388]: I don’t think it’s necessary to decompress again. 
return QCoreApplication::applicationFilePath() is okay

Comment (tamlok) [reply src/core/configmgr.cpp:387]: Yes, that is fine. I will use it only for the portable configuration folder.

**Comment by tamlok** [comment]
Thank @ygcaicn for the solution on Linux (https://github.com/vnotex/vnote/pull/1616).

**Pull request merged**